object-handlers.go

```diff
@@ -951,26 +951,26 @@ func (api objectAPIHandlers) CompleteMultipartUploadHandler(w http.ResponseWrite
 	w.WriteHeader(http.StatusOK)
 
 	//Send whitespace character, once every 5secs, until CompleteMultipartUpload is done
-	done := make(chan bool)
-	go func() {
+	doneCh := make(chan bool)
+	go func(doneCh <-chan bool) {
 		for {
 			select {
 			case <-time.After(5 * time.Second):
 				w.Write([]byte(" "))
```

Comment:
you can do this i think:
go func() {
    md5Sum, err = api.ObjectAPI.CompleteMultipartUpload(bucket, object, uploadID, completeParts)
    doneCh <- struct{}{}
}

for {
select{
    case <-time.After(5 * time.Second): w.Write([]byte(" "))
    case <-doneCh:break
}
}
if err != nil {
    writeErrorResponseNoHeader()
    return
}